controllers/events/event.go

```diff
@@ -0,0 +1,186 @@
+package events
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+
+	componentsv1alpha1 "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/json"
+	kservingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	openfunction "github.com/openfunction/apis/events/v1alpha1"
+	openfunctionevent "github.com/openfunction/apis/events/v1alpha1"
+)
+
+const (
+	EventSourceControlledLabel = "controlled-by-eventsource"
+	TriggerControlledLabel     = "controlled-by-trigger"
+
+	// EventSourceComponentNameTmpl => eventsource-{eventSourceName}-{sourceKind}-{eventName}
```

Comment:
The event source component name should be in format like below:
EventSourceComponentNameTmpl => eventsource-{sourceKind}-{eventSourceName}-{eventName}